Bottom: 5b3f255beb41807b0291155259c8f334859475d9
Top:    cf604fc7e07b8e8bf979b1ea69e9bc96a3002760
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-10-26 22:36:12 +0100

Temp fixes and diagnostics


---

diff --git a/src/polygon1.c b/src/polygon1.c
index 82baaf0..aa29d3d 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -1120,7 +1120,11 @@ label_contour (PLINE * a)
       assert (label == INSIDE || label == OUTSIDE);
       LABEL_NODE (cur, label);
     }
-  while ((cur = cur->next) != first_labelled);
+  while ((cur = cur->next) != first_labelled &&
+	 !(cur->next == &a->head && first_labelled == NULL));
+
+  if (first_labelled == NULL)
+    fprintf (stderr, "OOPS, DIDN'T LABEL ANYTHING!\n");
 #ifdef DEBUG_ALL_LABELS
   print_labels (a);
   DEBUGP ("\n\n");
diff --git a/src/sweep.h b/src/sweep.h
new file mode 100644
index 0000000..e69de29
